engine/src/ui/image/plugin.cpp

```diff
@@ -0,0 +1,126 @@
+#include <glm/glm.hpp>
+
+#include <cubos/engine/assets/plugin.hpp>
+#include <cubos/engine/render/shader/plugin.hpp>
+#include <cubos/engine/ui/canvas/draw_list.hpp>
+#include <cubos/engine/ui/canvas/element.hpp>
+#include <cubos/engine/ui/canvas/native_aspect_ratio.hpp>
+#include <cubos/engine/ui/canvas/plugin.hpp>
+#include <cubos/engine/ui/image/image.hpp>
+#include <cubos/engine/ui/image/plugin.hpp>
+#include <cubos/engine/window/plugin.hpp>
+
+using cubos::core::io::Window;
+
+using namespace cubos::engine;
+
+namespace
+{
+    struct PerElement
+    {
+        glm::vec2 xRange;
+        glm::vec2 yRange;
+        glm::vec4 color;
+        int depth;
```

Comment:
```suggestion
        int depth;
        float padding[3];
```
Necessary to work with WebGL (struct size must be a multiple of 16 bytes)